Source/Public/Get-SegmentedPath.ps1

```diff
@@ -41,49 +41,89 @@ function Get-SegmentedPath {
         [PoshCode.Pansies.RgbColor]$ErrorBackgroundColor
     )
 
-    $buffer = @()
+    # Define reusable constants.
+    $directorySeparatorChar = [System.IO.Path]::DirectorySeparatorChar
+    $ellipses = data { [char]0x2026 }
+    $resolvedHome = (Resolve-Path -LiteralPath $Home).Path
+    Write-Debug "`$directorySeparatorChar = '$directorySeparatorChar'"
+    Write-Debug "`$ellipses = '$ellipses'"
+    Write-Debug "`$resolvedHome = '$resolvedHome'"
+
+    # Resolve path.
+    $resolved = (Resolve-Path -LiteralPath $Path).Path
+    Write-Debug "`$resolved = '$resolved'"
+
+    # Split path into qualifier (or pseudo-qualifier) and no qualifier parts.
+    if($resolved.ToLower().StartsWith($resolvedHome.ToLower())) {
+        $qualifier = '~'
+        $noQualifier = $resolved.Substring($resolvedHome.Length)
+    } else {
+        $qualifier = Split-Path $resolved -Qualifier
+        $noQualifier = Split-Path $resolved -NoQualifier
+    }
+    Write-Debug "`$qualifier = '$qualifier'"
+    Write-Debug "`$noQualifier = '$noQualifier'"
+
+    # Ensure that the path neither starts nor ends with the directory separator character. This is necessary in
+    # order to avoid a reundant empty leading and/or trailing path segment when splitting the path into segments.
+    $trimmed = $noQualifier
+    if($trimmed.StartsWith($directorySeparatorChar)) {
+        $trimmed = $trimmed.Substring(1)
+    }
+    if($trimmed.EndsWith($directorySeparatorChar)) {
+        $trimmed = $trimmed.Substring(0, $trimmed.Length - 1)
+    }
+    Write-Debug "`$trimmed = '$trimmed'"
+
+    # Split path into segments.
+    $split = if($trimmed) { $trimmed.Split($directorySeparatorChar) } else { @() }
+    Write-Debug "`$split = @( $($split.ForEach{ "'$_'" } -join ', ') )"
+
+    #region Apply segment count and length limits.
+    $limited = @($qualifier) + $split
+    $prependEllipses = $false
+
+    function Test-SegmentLimit ([string[]] $Segments, [bool] $PrependEllipses) {
```

Comment:
We don't need to re-define the function every time Get-SegmentedPath is called...
In PowerLine, private functions go in \Private and don't have a - separator in the name.
You'll need to pass in the `$SegmentLimit` too.

If we're going to refactor this out, it should probably include the while loop, don't you think?